### ISSUE TITLE
Option resolveJsonModule doesn't work without esModuleInterop

### ISSUE THREAD

**Issue by ikokostya** [Microsoft/TypeScript#25400]
**TypeScript Version:**  2.9.2, 3.0.0-dev.20180703

**Search Terms:** 

resolveJsonModule, esModuleInterop

**Code**

```ts
import * as test from './test.json';
```

tsconfig.json:

```json
{
    "compilerOptions": {
        "module": "commonjs",
        "moduleResolution": "node",
        "target": "es2017",
        "lib": ["es2017"],
        "strict": true,
        "sourceMap": true,
        "noEmitOnError": true,
        "baseUrl": ".",
        "resolveJsonModule": true,
        "outDir": "out"
    }
}
```

**Expected behavior:**

No errors.

**Actual behavior:**

Compile error:

```
$ tsc
test.ts:1:23 - error TS2497: Module '"/home/kostya/tmp/resolve-json-test/test"' resolves to a non-module entity and cannot be imported using this construct.

1 import * as test from './test.json';
                        ~~~~~~~~~~~~~
```

***

When I add `esModuleInterop` option to `tsconfig.json`:

```diff
diff --git a/tsconfig.json b/tsconfig.json
index 7f1afb8..e949135 100644
--- a/tsconfig.json
+++ b/tsconfig.json
@@ -9,6 +9,7 @@
         "noEmitOnError": true,
         "baseUrl": ".",
         "resolveJsonModule": true,
+        "esModuleInterop": true,
         "outDir": "out"
     }
 }
```

and change import statement to:

```diff
diff --git a/test.ts b/test.ts
index 07bb9b7..dddcffb 100644
--- a/test.ts
+++ b/test.ts
@@ -1 +1 @@
-import * as test from './test.json';
+import test from './test.json';
```

it works. But with  `esModuleInterop` option I need to change many imports in my project, which is undesirable.

**Playground Link:**  

**Related Issues:**

**Comment by ikokostya** [comment]
Also with `resolveJsonModule`, but without `esModuleInterop`, json file appears in output directory.

**Comment by faltysad** [comment]
Confirmed that setting `resolveJsonModule` to `true` in my `tsconfig.json` resolved this issue.

**Comment by intelliot** [comment]
Manually restarting VS Code resolves this issue.

**Comment by ErnestoIslas** [comment]
> After setting resolveJsonModule to "true", I was having the same issue in VS Code, (TypeScript 3.2.2, VSCode 1.30.1). Restarting VS Code resolved the issue for me - I did not have to alter any other ts config

Thank you, it worked!

**Comment by arndwestermann** [comment]
> After setting resolveJsonModule to "true", I was having the same issue in VS Code, (TypeScript 3.2.2, VSCode 1.30.1). Restarting VS Code resolved the issue for me - I did not have to alter any other ts config

Worked for me as well, thanks!

**Comment by brionmario** [comment]
> After setting resolveJsonModule to "true", I was having the same issue in VS Code, (TypeScript 3.2.2, VSCode 1.30.1). Restarting VS Code resolved the issue for me - I did not have to alter any other ts config

This actually worked. Thanks 😃

**Comment by redi-monappab** [comment]
> After setting resolveJsonModule to "true", I was having the same issue in VS Code, (TypeScript 3.2.2, VSCode 1.30.1). Restarting VS Code resolved the issue for me - I did not have to alter any other ts config

Thanks, It worked like a charm!

**Comment by ltomes** [comment]
I see similar behavior in intellij to what @intelliot described above for vs code. Thanks for the suggestion.

**Comment by 0xEther** [comment]
For those who have updated their target to `esnext` you need to add the following to your **tsconfig**:
...
  "target"          : "esnext",
  "moduleResolution": "node",
  ...

**Comment by chrisj-skinner** [comment]
Initially after a restart the file is green, then 3 seconds later highlighted in red again. So restarting in my case doesn't fix this issue.

**Comment by imanuelgittens** [comment]
@chrisj-skinner same with me. Not sure why this is happening.

**Comment by NikosEfthias** [comment]
Why is this issue close since this problem still exists ?

**Comment by a-x-** [comment]
I have 
```
    "esModuleInterop": true,
    "resolveJsonModule": true,
```

and
`const pkgJSON = require('../../package.json');`
Consider using '--resolveJsonModule' to import module with '.json' extensionts(2732)

I using resolveJsonModule already.

**Comment by joofholland** [comment]
If you use WebStorm you shall also restart.

**Comment by Charada32** [comment]
I have the same issue here, 

> Initially after a restart the file is green, then 3 seconds later highlighted in red again. So restarting in my case doesn't fix this issue.

**Comment by axe-me** [comment]
> Initially after a restart the file is green, then 3 seconds later highlighted in red again. So restarting in my case doesn't fix this issue.

I had same issue, but that because I forgot to include the new created folder / ts file in the tsconfig.json

**Comment by clearloop** [comment]
> I had same issue, but that because I forgot to include the new created folder / ts file in the tsconfig.json

Same, thanks bro!

**Comment by davide1995** [comment]
In my case: "File" -> "Invalidate Caches / Restart..." -> "Invalidate and Restart", solved the issue

**Comment by danmalloc** [comment]
In my case, I fixed this by doing:```json
        "moduleResolution": "node",
```
This can also fix the issue:```json
        "module": "commonjs",
```

**Comment by oferRounds** [comment]
> Initially after a restart the file is green, then 3 seconds later highlighted in red again. So restarting in my case doesn't fix this issue.

@chrisj-skinner  can you explain how do it?

**Comment by oferRounds** [comment]
@axe-me 

> I had same issue, but that because I forgot to include the new created folder / ts file in the tsconfig.json

Can you please explain how to do it?

**Comment by SketchyLxve** [comment]
For my case, I'm unsure if this is suppose to happen or not but, I had to put my `tsconfig.json` in the same directory as my `out` and `src` folders.

**Comment by bujoriosif** [comment]
Although I added moduleResolution: node, the error was still there. After a VS restart it got 'fixed'

**Comment by heavenkiller2018** [comment]
I have the same problem. all of "moduleResolution: node", "resolveJsonModule":true and "module": "commonjs" have no effect! for help!

**Comment by Miciurash** [comment]
Thank you @bujoriosif ! your "fix" worked for me :)

**Comment by sdseenu** [comment]
Stil i got same issues do it all steps above but noting to work. Any idea
import * as data from '../e2e/src/testData/login.json';
![image](https://user-images.githubusercontent.com/25522319/109956537-c5b3ba80-7d09-11eb-8341-206d04279806.png)
import error
![image](https://user-images.githubusercontent.com/25522319/109956613-d82df400-7d09-11eb-91d3-4ec781ed868d.png)

Anyone to help

**Comment by bujoriosif** [comment]
> Stil i got same issues do it all steps above but noting to work. Any idea
> import * as data from '../e2e/src/testData/login.json';
> ![image](https://user-images.githubusercontent.com/25522319/109956537-c5b3ba80-7d09-11eb-8341-206d04279806.png)
> import error
> ![image](https://user-images.githubusercontent.com/25522319/109956613-d82df400-7d09-11eb-91d3-4ec781ed868d.png)
> 
> Anyone to help

Dummy question maybe: did you restarted VS Code after adding/changing moduelResolution? 
Also resolveJsonModule is false,  based on experience and answers above, it should be true.  
Also try changing target to 'esNext'.

**Comment by sdseenu** [comment]
@bujoriosif  is done all but not working 
![image](https://user-images.githubusercontent.com/25522319/109959051-f9441400-7d0c-11eb-83d7-040191bf0559.png)

**Comment by sdseenu** [comment]
Any one to suggest am able to import json file. unfortunately not working

**Comment by bujoriosif** [comment]
> Any one to suggest am able to import json file. unfortunately not working

what is the more specific error now? (as I understood, the previous one regarding import is gone now)

**Comment by sdseenu** [comment]
> > Any one to suggest am able to import json file. unfortunately not working
> 
> what is the more specific error now? (as I understood, the previous one regarding import is gone now)

This is error 
![image](https://user-images.githubusercontent.com/25522319/109966401-3c56b500-7d16-11eb-9385-73b3f14427a5.png)

**Comment by sdseenu** [comment]
> > > Any one to suggest am able to import json file. unfortunately not working
> > 
> > 
> > what is the more specific error now? (as I understood, the previous one regarding import is gone now)
> 
> This is error
> ![image](https://user-images.githubusercontent.com/25522319/109966401-3c56b500-7d16-11eb-9385-73b3f14427a5.png)

![image](https://user-images.githubusercontent.com/25522319/109966648-863f9b00-7d16-11eb-81f3-90d2cb453206.png)

**Comment by cdpautsch** [comment]
I am running into the same issue. None of the suggested fixes work for me:

My `tsconfig.json`:
```
"target": "esnext",    
"moduleResolution": "node",
"esModuleInterop": true,
"resolveJsonModule": true
```

I have reloaded and restarted but I am still getting the error.

**Comment by MohammedFarooqS** [comment]
In tsconfig.json
{
"compilerOptions": {
"target": "esnext",
"lib": [
"dom",
"dom.iterable",
"esnext"
],
"allowJs": true,
"skipLibCheck": true,
"esModuleInterop": true,
"allowSyntheticDefaultImports": true,
"strict": true,
"forceConsistentCasingInFileNames": true,
"module": "esnext",
"moduleResolution": "node",
"resolveJsonModule": true,
"isolatedModules": true,
"noEmit": true,
"jsx": "preserve"
},
"include": [
"src"
]
}

Npm install
Restart VSCODE

**Comment by Ibrahim-Mohamed2100** [comment]
Added these two flags Fixed it,
```
"compilerOptions": {
"resolveJsonModule": true,
"allowSyntheticDefaultImports":true
}
```

Hope to Help!

**Comment by shivamjjha** [comment]
See [Importing JSON file in TypeScript](https://stackoverflow.com/a/50674344/11667949)

**Comment by hamidhadi** [comment]
I still have this issue, and none of the mentioned solutions works for me.

**Comment by edandweb** [comment]
I add in the include `"./**/*.json"`
My include: `  "include": [
    "./**/*", "./**/*.json"
  ],`


For me it did the job :).

**Comment by tncrazvan** [comment]
Restarting works as mentioned already, but you don't have to actually restart VSCode, you can just restart type script itself:
![image](https://user-images.githubusercontent.com/6891346/151166544-b60a71fb-f76d-4b24-ae7a-e33a944665d4.png)

**Comment by michaelwilcox** [comment]
+1 restarting VSCode after adding `"resolveJsonModule": true` to `tsconfig.json` solved issue for me.

**Comment by kahilkubilay** [comment]
> +1 restarting VSCode after adding `"resolveJsonModule": true` to `tsconfig.json` solved issue for me.

Could you share screenshots of `tsconfig` and `rollup.config.js` files?

In combination with `tsconfig`, maybe setting `"resolveJsonModule": true` in `rollup.config.js` will solve your problem.

**Comment by rubnawazqureshidev** [comment]
> > After setting resolveJsonModule to "true", I was having the same issue in VS Code, (TypeScript 3.2.2, VSCode 1.30.1). Restarting VS Code resolved the issue for me - I did not have to alter any other ts config
> 
> This actually worked. Thanks 😃

I same.

**Comment by dgallardox** [comment]
> After setting resolveJsonModule to "true", I was having the same issue in VS Code, (TypeScript 3.2.2, VSCode 1.30.1). Restarting VS Code resolved the issue for me - I did not have to alter any other ts config

Same here. A simple restart fixed it. Thanks!

**Comment by bzayn02** [comment]
Adding "moduleResolution":"Node" in tsconfig.json file, can help fix this issue.

**Comment by EdsonAOJ** [comment]
I just add the code 
```
  "moduleResolution": "node",
```
and it stop to get error.

**Comment by AlexHelmutSonntag** [comment]
I restarted the TS Server and it did not help. Here are my tsconfig.json and my ts file.
![Screenshot from 2022-07-06 17-07-49](h
![Screenshot from 2022-07-06 17-08-06](https://user-images.githubusercontent.com/61309161/177583402-01105a0f-03a3-4168-b410-4d5e53b7e3e7.png)
ttps://user-images.githubusercontent.com/61309161/177583394-694ab376-7ded-40ed-935f-1627215d2488.png)

**Comment by ThayllanVivas** [comment]
I added     "moduleResolution": "node" and it worked

**Comment by PrinceBrown** [comment]
> For those who have updated their target to `esnext` you need to add the following to your **tsconfig**: ... "target" : "esnext", "moduleResolution": "node", ...

This actually works when your
 "target": "esNext" 
Thank you

**Comment by Michael-Swordys** [comment]
> Depois de definir resolveJsonModule como "true", eu estava tendo o mesmo problema no VS Code (TypeScript 3.2.2, VSCode 1.30.1). Reiniciar o VS Code resolveu o problema para mim - não precisei alterar nenhuma outra configuração ts

Isso resolveu meu problema, obrigado!

**Comment by yk-curu2** [comment]
I could not imoprt the package.json file to the vite.config.ts file.
I solved the problem by editing `tsconfig.node.json` as follows.

```json
{
  "compilerOptions": {
    "composite": true,
    "module": "ESNext",
    "moduleResolution": "Node",
    "resolveJsonModule": true, // add
    "allowSyntheticDefaultImports": true
  },
  "include": [
    "vite.config.ts",
    "package.json" // add.
  ]
}
```

`vite.config.ts`
```ts
import pkg from 'package.json'.
```

Finally, run it in VSCode's command palette.

`Typescript: Restart TS server`

We hope this helps.

**Comment by avendesta** [comment]
Okay, this is weird. I'm on Mac and using Vite to create my React app. I tried most of the above but none worked. I uninstalled VSCode, upgrade typescript, recreated my project and it worked.

**Comment by bamoha** [comment]
Updating VScode and restarting it fixed the issue for me.

**Comment by zahidshowrav** [comment]
In my Nextjs (v13.5.3) Typescript(v5.2.2) project, I had to keep `"moduleResolution": "bundler"`, but I was getting the same error. Then I got that my vs code was running an older version of Typescript(v4.8.2). I have changed the vs code's Typescript use version to my workspace(V5.2.2). **Bingoo**... error gone :).

Following is how to change the vs code Typescript use version in 3 steps:
1. Opening the command palette (Ctrl/⌘ + Shift + P)
2. Searching for "TypeScript: Select TypeScript Version"
3. Selecting "Use Workspace Version" 

Screenshot (Step 2):
![Screenshot from 2023-09-30 15-55-23](https://github.com/microsoft/TypeScript/assets/26140192/2e5edd43-d24f-40c0-90ab-7b3b7f13f70e)

 Screenshot (Step 3):
![Screenshot from 2023-09-30 15-20-12](https://github.com/microsoft/TypeScript/assets/26140192/a46e6d81-c3c2-43fa-8c3a-db51c7dad333)
